### PR TITLE
.github/workflows: work around golang/go#51629

### PR DESCRIPTION
**Pull request by danderson**
Updates #4194

Signed-off-by: David Anderson <danderson@tailscale.com>

### CODE REVIEW

**Comment by maisem** [comment]
I disabled the check yesterday, will have to enable it again from the UI

**Comment by danderson** [comment]
Confused: we have a workflow for go_generate in .github/workflows, but it doesn't seem to run in CI?...

**Comment by maisem** [comment]
I just renabled it, want to push again?

File: .github/workflows/go-generate-without-stringer.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# This is a temporary hack to work around
+# https://github.com/golang/go/issues/51629 , wherein the stringer
+# generator doesn't work with generics.
+#
+# This script is the equivalent of `go generate ./...`, except that it
+# only runs generate on packages that don't try to use stringer.
+
+find . -name '*.go' | xargs grep -l go:generate | xargs -n1 dirname | sort -u | while read dir; do
+	if ! grep cmd/stringer $dir/*.go; then
+		set -x
```
Comment (maisem) [inline .github/workflows/go-generate-without-stringer.sh:12]: Is this intended to be equivalent to `go generate || true`?

**Review by josharian — APPROVED** [review]

File: .github/workflows/go-generate-without-stringer.sh (hunk 1 above)
Comment (danderson) [reply .github/workflows/go-generate-without-stringer.sh:12]: Sort of. My intent is to skip go generate tasks that we know are going to fail, while still reporting failures for others. What I appear to have discovered, is that this workflow never complained if `go generate` fails (vs. succeeds but produces a diff), oops. Working on it.

File: net/dnsfallback/update-dns-fallbacks.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ import (
 )
 
 func main() {
+	log.Printf("RUN RUN RUN")
```
Comment (maisem) [inline net/dnsfallback/update-dns-fallbacks.go:22]: Debug?

**Review by maisem — APPROVED** [review]

File: net/dnsfallback/update-dns-fallbacks.go (hunk 2 above)
Comment (danderson) [reply net/dnsfallback/update-dns-fallbacks.go]: Oops! Fixed.

**Pull request merged**